crates/adapters/src/transport/kafka/nonft/input.rs

```diff
@@ -146,7 +151,10 @@ impl KafkaInputReaderInner {
                 if let Some(payload) = message.payload() {
                     // Leave it to the controller to handle errors.  There is noone we can
                     // forward the error to upstream.
-                    let _ = consumer.input_chunk(payload);
+                    let _ = parser.input_chunk(payload);
+                    if let Some(buffer) = parser.take() {
+                        self.queue.lock().unwrap().push_back(buffer);
```

Comment:
This won't help this connector scale :( I guess the synchronization point was always there, just hiding behind a few layers of code.